simba/report.py

```diff
@@ -267,6 +291,268 @@ def generate(schedule, scenario, args):
     logging.info(f"Plots and output files saved in {args.results_directory}")
 
 
+def bus_type_distribution_route_rotation(extended_plots_path, schedule):
+    """Plots the distribution of bus types in consumption brackets as a stacked bar chart.
+
+    :param extended_plots_path: Path argument which combines args.results_directory
+                                and string 'extended_plots'
+    :type extended_plots_path: PosixPath
+    :param schedule: Driving schedule for the simulation. schedule.rotations are used
+    :type schedule: eBus-Toolbox.Schedule
+    """
+
+    step = 50
```

Comment:
Constant step size declared in function? Should this be a setting for the user? Is it just for testing and pyplot might make reasonable steps on its own?

Reply:
I think the following change would make sense:

- Guarantee a minimum amount of bins

Optional:
- Do not plot empty bins between 0 and minimal consumption

Reply:
Changed: dynamic step size / number of bins. Number of bins is always between 1 and 20, optimally (number of rotations) / 2. Step size goes 1, 2.5, 5, 10, 25, 50, 100, 250, ...
Bins below minimum value are neither counted nor displayed.